src/templates/default.html

```diff
@@ -39,9 +39,57 @@
       <link rel="stylesheet" href="./css/home.css" />
     $endif$
   </head>
-  <body class="dark preload">
+  <body data-theme="dark">
+    <script>
+      ;(() => {
+        window.site = {
+          theme: localStorage.getItem('theme') || (matchMedia('(prefers-color-scheme: dark)').matches ? 'dark' : 'light'),
+          setTheme: (name) => {
+            document.body.setAttribute('data-theme', name)
+            localStorage.setItem('theme', name);
+            window.site.theme = name;
+            window.site.refreshThemeSwitch();
+
+            document.querySelector('iframe.giscus-frame')?.contentWindow.postMessage(
+              { giscus: { setConfig: { theme: name === 'dark' ? 'dark_dimmed' : 'light' } } },
+              'https://giscus.app'
+            );
```

Comment:
I didn't verify if this works on the posts, so you might want to double check...it should! If it doesn't, I'm sure you can figure it out.